src/plugin_discovery/errors.js

```diff
@@ -0,0 +1,63 @@
+
+const $code = Symbol('FindErrorCode');
+
+/**
+ *  Thrown when reading a plugin directory fails, wraps failure
+ *  @type {String}
+ */
+const ERROR_INVALID_DIRECTORY = 'ERROR_INVALID_DIRECTORY';
+export function createInvalidDirectoryError(sourceError, path) {
+  sourceError[$code] = ERROR_INVALID_DIRECTORY;
+  sourceError.path = path;
+  return sourceError;
+}
+export function isInvalidDirectoryError(error) {
+  return error && error[$code] === ERROR_INVALID_DIRECTORY;
+}
+
```

Comment:
Just a general comment about this file. I like that the errors, error creators, and selectors are in the same file. I hadn't used that approach before, and it makes sense.